### ISSUE TITLE
eye landmarkss initialization

### ISSUE THREAD

**Issue by chengtingting** [TadasBaltrusaitis/OpenFace#715]
hello, it is me again! I want to replace the 6 eye initial landmarks with my own eye 2d landmarks in image. but the final fitted 28 eye landmarks is very different with your output. my eye landmarks has a bit of bias with your 6 eye landmarks from facial landmarks. Would you tell me why the initial landmakrs has so many impacts ?

**Comment by TadasBaltrusaitis** [comment]
Hi,

Sorry I'm not entirely sure I understand the question. There are two sets of eye landmarks that are output by OpenFace - the 6 "coarse" landmarks and 28 "dense" landmarks. To initialize the detection of the dense landmarks the coarse set is used.

**Comment by chengtingting** [comment]
@TadasBaltrusaitis yes, i mean the CLNF of eye is sensitive to the 6 coarse landmakrs. For example, the eye 28 landmarks become bad when I add or sub several pixels to the 6 coarse landmarks and then use the changed ones as initialization. Why will come this phenomenon ? Is the ability of refining the eye landmarks is very limited?

**Comment by TadasBaltrusaitis** [comment]
The 28 landmark detector models were trained specifically with the accuracy of the coarse 6 landmarks in mind, so providing less accurate landmarks would lead to suboptimal results. Further, they are trained at larger scale so would be quite sensitive even to small pixel variations (this was done partly because the training data was synthetic and only a part of the face could be rendered). Hope this answers your question.
